src/lavinmq/queue/stream_queue.cr

```diff
@@ -0,0 +1,187 @@
+require "./durable_queue"
+
+module LavinMQ
+  class StreamQueue < Queue
+    @durable = true
+    @exclusive_consumer = false
+    @no_ack = true
+
+    private def init_msg_store(data_dir)
+      @msg_store = StreamQueueMessageStore.new(data_dir)
+    end
+
+    class StreamQueueMessageStore < MessageStore
```

Comment:
You need a read segment/position per consumer, unlike the normal message store.

Reply:
Maybe a MessageStore per consumer? is that possible?

Reply:
that will create duplicate `MFile`s, a bit of waste, but maybe not too bad, can try it out. otherwise somehow store the current Segment and Position per consumer, and with each shift update it.

Reply:
also need a sepearate `@requeue` per consumer.

Reply:
Yeah, not sure what the best solution is yet. I think storing Segment and Position per consumer will be easiest. But we also need a good way to "seek" to an offset I guess? If a new consumer starts consuming at a random offset.

Reply:
Consumers can't "re-consume specific offsets", then the consumer has to be cancceled and start a new consumer with a certain offset, all other prefetched messages the conusmer might have received have to be redelivered then. So requeueing support is required imho. 

We should not create a new queue per consumer, that won't scale very well. Better that each consumer "remebers" where it is is in the queue.